frontend/components/tree/js/tree.service.ts

```diff
@@ -1034,36 +1045,38 @@ export class TreeService {
 		this.ViewerService.clearHighlights();
 		this.DocsService.closeDocs();
 
-		const visitedParents = [];
-		const visitedChildren = [];
+		const visitedNodes = {};
 
-		while (this.currentSelectedNodes.length) {
-			const currentNode = this.currentSelectedNodes.pop();
-			currentNode.selected = this.SELECTION_STATES.unselected;
+		for (const id in this.currentSelectedNodes) {
+
+			if (!id) { continue; }
```

Comment:
this id is the same as currentNode._id right?

should just merge L1057 with this. ie:

````js
if(!id || visitedNodes[currentNode._id]) { continue; }
````
Should save some operation as iterations where they fall into the continue line at L1058 wouldnt perform the look up on L1054. Also should avoid multiple `continue` code where possible